src/cli/generate-idlnames.js

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env node
+/**
+ * The IDL names generator takes a crawl report as input and creates a report
+ * per referenceable IDL name, that details the complete parsed IDL structure
+ * that defines the name across all specs.
+ *
+ * The spec checker can be called directly through:
+ *
+ * `node generate-idlnames.js [crawl report] [save folder]`
+ *
+ * where `crawl report` is the path to the folder that contains the
+ * `index.json` file and all other crawl results produced by crawl-specs.js,
+ * and `save folder` is an optional folder (which must exist) where IDL name
+ * extracts are to be saved. In the absence of this parameter, the report is
+ * written to the console.
+ *
+ * @module checker
+ */
+
+const fs = require('fs');
+const path = require('path');
+const { requireFromWorkingDirectory, expandCrawlResult } = require('../lib/util');
+const { matchIdlDfn, getExpectedDfnFromIdlDesc } = require('./check-missing-dfns');
+
+
+function linkifyIdl(idl, results) {
+  const spec = results.find(s => s.url === idl.spec.url);
+  if (!spec || !spec.dfns) {
+    return;
+  }
+
+  function traverseIdl(idl, parentIdl) {
```

Comment:
probably minor, but the way the code is structured it's a bit hard to identify at a glance what the function actually does (the fact that the main action is in line 6 of the body which is neither the start nor the end probably contributes to this).

The function is also pseudo-recursive (only one level of recursion AFAICT) - it may be easier to read if it was flattened?

Reply:
thanks, this reads much better!